lib/core/src/restore.rs

```diff
@@ -29,135 +29,183 @@ impl TxMap {
-    btc_user_lockup_tx_id: Option<HistoryTxId>,
+    pub(crate) btc_user_lockup_tx_id: Option<HistoryTxId>,
     /// BTC tx initiated by the SDK to a user-chosen address, in case the initial funds have to be refunded.
-    btc_refund_tx_id: Option<HistoryTxId>,
+    pub(crate) btc_refund_tx_id: Option<HistoryTxId>,
 }
 impl PartialSwapState for RecoveredOnchainDataChainReceive {
-    fn derive_partial_state(&self) -> PaymentState {
+    fn derive_partial_state(&self) -> Option<PaymentState> {
         match &self.btc_user_lockup_tx_id {
             Some(_) => match &self.lbtc_server_claim_tx_id {
                 Some(lbtc_server_claim_tx_id) => match lbtc_server_claim_tx_id.confirmed() {
-                    true => PaymentState::Complete,
-                    false => PaymentState::Pending,
+                    true => Some(PaymentState::Complete),
+                    false => Some(PaymentState::Pending),
                 },
                 None => match &self.btc_refund_tx_id {
                     Some(tx) => match tx.confirmed() {
-                        true => PaymentState::Failed,
-                        false => PaymentState::RefundPending,
+                        true => Some(PaymentState::Failed),
+                        false => Some(PaymentState::RefundPending),
                     },
-                    None => PaymentState::Created,
+                    None => Some(PaymentState::Pending),
                 },
             },
-            None => PaymentState::Created,
+            // We have no onchain data to support deriving the state as the swap could
+            // potentially be Created or Failed after expiry. In this case we return None.
+            None => None,
         }
     }
 }
 
 pub(crate) struct RecoveredOnchainData {
-    send: HashMap<String, RecoveredOnchainDataSend>,
-    receive: HashMap<String, RecoveredOnchainDataReceive>,
-    chain_send: HashMap<String, RecoveredOnchainDataChainSend>,
-    chain_receive: HashMap<String, RecoveredOnchainDataChainReceive>,
+    pub(crate) send: HashMap<String, RecoveredOnchainDataSend>,
+    pub(crate) receive: HashMap<String, RecoveredOnchainDataReceive>,
+    pub(crate) chain_send: HashMap<String, RecoveredOnchainDataChainSend>,
+    pub(crate) chain_receive: HashMap<String, RecoveredOnchainDataChainReceive>,
 }
 
 impl LiquidSdk {
+    pub(crate) async fn get_monitored_swaps_list(&self, partial_sync: bool) -> Result<SwapsList> {
+        let receive_swaps = self.persister.list_ongoing_receive_swaps()?;
+        match partial_sync {
+            false => {
+                let bitcoin_height = self.bitcoin_chain_service.lock().await.tip()?.height as u32;
+                let liquid_height = self.liquid_chain_service.lock().await.tip().await?;
+                let final_swap_states = [PaymentState::Complete, PaymentState::Failed];
+
+                let send_swaps = self.persister.list_pending_send_swaps()?;
```

Comment:
cc @roeierez - Turns out we need to track `Created`, `Pending` and `RefundPending` in order for the real-time sync to work as expected. Either we create a new persister method to encapsulate the three (`list_pending_and_ongoing_send_swaps`), or I could also set the initial state for a synced send swap to `Pending`. Does that makes sense from a UX perspective?